### ISSUE TITLE
Impossible to use $httpBackend.expect() when using HTTP mocks from ng-describe?

### ISSUE THREAD

**Issue by jakub-g** [kensho/ng-describe#79]
Hi, 

we have in our test the following code to serve mocked responses:

```
ngDescribe({
...
  inject: [..., '$httpBackend',],
  mocks: {
    ng: {
      $http: {
        post: function($q, url) {
          return $q.when({
            data: JSON.parse(readMock(url)) // `readMock` does a synchronous XHR to a mock JSON file
          });
        }
      }
    }
  },
```

then I have a test case as follows (I want to make sure that a POST request is sent)

```
it('contacts the server when input is XYZ', function() {
  deps.$httpBackend.expect("POST", "foobar", payload);

  var wasPromiseOk = false;
  deps.MyService.doSomething("XYZ").then(function() {
    wasPromiseOk = true;
  }, function errorCb() {
    wasPromiseOk = false;
  });

  deps.$httpBackend.flush();
  deps.$rootScope.$digest();

  expect(wasPromiseOk).toBe(true);
  deps.$httpBackend.verifyNoOutstandingExpectation();
}
```

The function under test (`doSomething`)  does the call

     $http.post('foobar', payload);

However the test fails

> Error: Unsatisfied requests: POST foobar

However when I remove ngdescribe mocks, and define a mock response manually instead, the test is ok:

     deps.$httpBackend.expect("POST", url, payload).respond(getMockResponder(url));

Digging into the issue it seems related to this SO thread

http://stackoverflow.com/questions/17858955/how-to-override-mocked-response-of-httpbackend-in-angular

In short, it seems that when ngdescribe mocks are instantiated, they use `$httpBackend.when` and this can not be overridden later on using another   `$httpBackend.when` but also  `$httpBackend.expect` does not register passing through the earlier-defined mocks.

Not sure if something can be done about it, maybe the docs on the mocks can be improved to let others know about this issue?
